### PR TITLE
DPD-BASIC kokkosification 

### PR DESCRIPTION
**Pull request by bathmatt**
**Summary**

Developed Kokkos versions of the four routines in DPD-BASIC

**Author(s)**

Developed by Matt Bettencourt mattb@nvidia.com  Internal review by Evan Weinberg eweinberg@nvidia.com

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

no

**Implementation Notes**

Set the RNG within the modules and confirmed that the results were identical for the first many steps, 

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] The feature has been verified to work with the conventional build system

**Further Information, Files, and Links**

### CODE REVIEW

File: src/DPD-BASIC/pair_dpd.h [hunk 1]
```diff
@@ -28,11 +28,11 @@ class PairDPD : public Pair {
  public:
   PairDPD(class LAMMPS *);
   ~PairDPD() override;
-  void compute(int, int) override;
+  virtual void compute(int, int) override;
```
Comment (akohlmey) [inline src/DPD-BASIC/pair_dpd.h:31]: Why add "virtual" here? The "override" should be sufficient.

Comment (bathmatt) [reply src/DPD-BASIC/pair_dpd.h:31]: Habit mostly, I can remove all the virtuals in the DPD dir, I typically use virtual if someone builds on top of it.

Comment (akohlmey) [reply src/DPD-BASIC/pair_dpd.h:31]: As far as I understood the standard, functions don't lose their polymorphic status when "override" is used (they do without). So "virtual" is needed for the first definition and then "override" for all following unless you use "final" to prohibit further overrides.

Comment (bathmatt) [reply src/DPD-BASIC/pair_dpd.h:31]: That is the way I understand it as well.  I will remove the added virtuals as that is the style of LAMMPS as of a bit ago.

Comment (akohlmey) [reply src/DPD-BASIC/pair_dpd.h:31]: Perhaps some background info is helpful: we've run the code through clang-tidy to modernize some common constructs and would like to keep all new code consistent with that, so we don't have to do the clang-tidy processing all the time. :wink:

Comment (stanmoore1) [reply src/DPD-BASIC/pair_dpd.h:31]: >So "virtual" is needed for the first definition and then "override" for all following unless you use "final" to prohibit further overrides.

I wasn't aware of this--will keep this in mind for future code as well.

Comment (bathmatt) [reply src/DPD-BASIC/pair_dpd.h]: I pushed code without the virtual where it isn't needed, it is still needed in allocate cuz that is the start of the chain.

**Comment by bathmatt** [comment]
@stanmoore1 @akohlmey how does this look?  tutti bene?  wanna close it out on my end if it is good else fix things before I forget.
Thanks!

**Comment by akohlmey** [comment]
> @stanmoore1 @akohlmey how does this look? tutti bene? wanna close it out on my end if it is good else fix things before I forget. Thanks!

@bathmatt not yet *tutto bene*, but it is currently in the hands of @stanmoore1 and he'll eventually get to check it out.
As far as I am concerned, the things that need to be done are just minor changes to build system, documentation, etc. to fully integrate your changes. Those modifications are easier to do myself than to explain. So unless you explicitly ask be to point them all out and explain, I'll just wait and do them after the science has been checked and confirmed.

**Comment by stanmoore1** [comment]
@bathmatt this is really close. I just need to do final checks and also need a few updates to the docs.

**Comment by stanmoore1** [comment]
@akohlmey I think it would be good to add an example for `pair_dpd` and `pair_dpd_tstat`.

**Comment by akohlmey** [comment]
> @akohlmey I think it would be good to add an example for `pair_dpd` and `pair_dpd_tstat`.

done

**Comment by bathmatt** [comment]
Let me know what I can do to help out.

**Comment by stanmoore1** [comment]
This passes all my tests now.

**Comment by stanmoore1** [comment]
@akohlmey this PR is ready to merge from my point of view.

**Review by sjplimp — APPROVED** [review]

**Pull request merged**